pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -203,6 +204,28 @@ func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
 			}
 			cfg[url] = *cred
 		}
+
+		// Handle the custom cloud case
+		// In clouds where ACR is not yet deployed, the string will be empty
+		if strings.Contains(a.environment.ContainerRegistryDNSSuffix, ".azurecr.") {
+			customAcrSuffix := "*" + a.environment.ContainerRegistryDNSSuffix
+			hasBeenAdded := false
+			for _, url := range containerRegistryUrls {
+				if url == customAcrSuffix {
+					hasBeenAdded = true
+					break
+				}
+			}
+
+			if !hasBeenAdded {
+				cred := &credentialprovider.DockerConfigEntry{
+					Username: a.config.AADClientID,
```

Comment:
there is a security issue here:
if we support self defined ACR domain, e.g. `test.azurecr.my.cloud`, then service principal secret would be sent to that remote server. 
The reason why we use hardcoded domains is that we could always make sure the following 4 domains are owned by Azure:
https://github.com/kubernetes/kubernetes/blob/0c0984abdb17246b472e65f0fefee82637dd83e6/pkg/credentialprovider/azure/azure_credentials.go#L49-L50

Reply:
The environment is fetched from ARM. Thus, the only acr endpoints used here will be the one returned by ARM. This needs to be allowed in order to support clouds with urls we are not allowed to expose publicly.

Reply:
that sounds reasonable, thanks.